reclist/metrics/standard_metrics.py

```diff
@@ -127,4 +133,62 @@ def precision_at_k(y_preds, y_test, k=3):
 
 def recall_at_k(y_preds, y_test, k=3):
     recall_ls = [len(set(_y).intersection(set(_p[:k]))) / len(_y) if _y else 1 for _p, _y in zip(y_preds, y_test)]
-    return np.average(recall_ls)
\ No newline at end of file
+    return np.average(recall_ls)
+
+
+def rec_items_distribution_at_k(y_preds, k=3, bin_width=100, debug=True):
+    def _calculate_hist(frequencies, bins):
+        """ Works out the counts of items in each bucket """
+        counts_per_bin = Counter([bisect_right(bins, item) - 1 for item in frequencies])
+        counts_per_bin_list = list(counts_per_bin.items())
+        empty_bins_indices = [ele for ele in np.arange(len(bins) - 1) if ele not in [
+            index for index, _ in counts_per_bin_list
+        ]]
+        counts_per_bin_list.extend([(index, 0) for index in empty_bins_indices])
+        counts_per_bin_sorted = sorted(counts_per_bin_list, key=lambda x: x[0], reverse=False)
+        return [y for _, y in counts_per_bin_sorted]
```

Comment:
Nice!

Reply:
That was painful to compute manually! I appreciated the existence of libraries 😅